support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,283 @@
+package org.apereo.cas.ticket.registry;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import org.apereo.cas.TicketSerializer;
+import org.apereo.cas.ticket.ServiceTicket;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+import org.apereo.cas.ticket.TicketGrantingTicketImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * @author David Rodriguez
+ *
+ * @since 5.1.0
+ */
+public class CassandraTicketRegistry<T> extends AbstractTicketRegistry implements CassandraTicketRegistryDao {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraTicketRegistry.class);
+
+    private static final int FIRST_COLUMN_INDEX = 0;
+    private static final long TEN_SECONDS = 10000L;
+    private static final int TEN = 10;
+
+    private final TicketSerializer<T> serializer;
+    private final Class<T> typeToWriteToCassandra;
+
+    private final PreparedStatement insertTgtStmt;
+    private final PreparedStatement updateTgtStmt;
+    private final PreparedStatement selectTgtStmt;
+    private final PreparedStatement deleteTgtStmt;
+
+    private final PreparedStatement insertStStmt;
+    private final PreparedStatement updateStStmt;
+    private final PreparedStatement selectStStmt;
+    private final PreparedStatement deleteStStmt;
+
+    private final PreparedStatement selectExStmt;
+    private final PreparedStatement selectDateExStmt;
+
+    private final PreparedStatement selectLrStmt;
+    private final PreparedStatement updateLrStmt;
+
+    private final Session session;
+
+    public CassandraTicketRegistry(final String contactPoints, final String username, final String password, final TicketSerializer<T> serializer,
+                                   final Class<T> typeToWriteToCassandra, final String tgtTable, final String stTable, final String expiryTable,
+                                   final String lastRunTable) {
+        this.serializer = serializer;
+        this.typeToWriteToCassandra = typeToWriteToCassandra;
+        final Cluster cluster = Cluster.builder().addContactPoints(contactPoints.split(",")).withCredentials(username, password)
+                .withProtocolVersion(ProtocolVersion.V3).build();
+
+        this.session = cluster.connect();
+
+        this.selectTgtStmt = session.prepare("select ticket from " + tgtTable + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
```

Comment:
This also will not work, with the introduction of additional ticket types that exist today. You want to discover ticket types, and auto-create tables, etc based on what exists at runtime and has been registered. Not statically, and without any assumptions.